inc/ResponseBuilder.hpp

```diff
@@ -29,11 +29,10 @@ class ResponseBuilder {
 	void appendHeaders(const HTTPRequest& request);
 	std::string getMIMEType(const std::string& extension);
 	void initMIMETypes();
-	void resetStream();
+	void resetBuilder();
 
 	std::map<std::string, std::string> m_mimeTypes;
 	const FileSystemPolicy& m_fileSystemPolicy;
-	std::stringstream m_responseStream;
+	std::stringstream m_responseHeader;
 	std::string m_responseBody;
```

Comment:
So the std::string works for binary data if the << and += operators are avoided?

Reply:
The root problem is that binary data doesn't work like c-strings: they can contain "zero terminators" aka a zero byte in the middle of the data.

std::string acts like a std::vector for char (bytes) but for compatibility for "old" c-strings also has various functions which work on a const char * (which is expected to be zero terminated). This can lead to bugs if invoked on data which holds non-string data (binary data). One has to be careful not to invoke the wrong function. 
So std::string works but actually "isn't the right tool".

Std::stringstream has the same "problem": it's a stream class to operate on strings. The << operator works on arithmetic types (const char *) and not on std::string. Which leads to the same problem with binary data. We could work around this by constructing a stream out of the std::string and insert the stream buffer (which should insert it "as is").

So both containers work with binary data, but you have to be careful to not invoke functions which work on const char *.

I chose to use stringstream for response header as it is text = string.
I kept std::string to keep consistent with other parts of our code.
I think std::vector<unsigned char> would be the "correct" container to choose when handling binary data.

Reply:
Thanks for the clarification. Do I understand you correctly that ftm we stay with the std::string and try to avoid functions that utilise a const char* and have no intention of switching to std::vector yet (except for it becomes really unavoidable)?

Reply:
Yes, I would leave std::string where it's already in codebase and works. But we should keep an eye out for unintentional usages for const char *

If we add new buffers, I propose we prefer std::vector - though there shouldn't be much "opportunities" left.